src/main/java/io/openmessaging/storage/dledger/DLedgerEntryPusher.java

```diff
@@ -396,8 +397,10 @@ private void checkQuotaAndWait(DLedgerEntry entry) {
             }
         }
         private void doAppendInner(long index) throws Exception {
-            DLedgerEntry entry = dLedgerStore.get(index);
-            PreConditions.check(entry != null, DLedgerResponseCode.UNKNOWN, "writeIndex=%d", index);
+            DLedgerEntry entry = getDLedgerEntryForAppend(index);
+            if (null == entry) {
+                return;
+            }
```

Comment:
Is this useless code？ If executed here, entry does not seem to be null.

Reply:
In line 443, with response code is INDEX_LESS_THAN_LOCAL_BEGIN, it can be null.
I want it just return to do compare.

Reply:
I see

Reply:
> In line 443, with response code is INDEX_LESS_THAN_LOCAL_BEGIN, it can be null.
> I want it just return to do compare.

I see